### PR TITLE
{Packaging} Fix jp download url in ARM64 docker image

### PR DESCRIPTION
**Pull request by bebound**
**Description**

The url link in ARM64 should be updated.

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Packaging

File: Dockerfile [hunk 1]
```diff
@@ -42,9 +42,9 @@ RUN apk add --no-cache bash openssh ca-certificates jq curl openssl perl git zip
  && apk add --no-cache bash-completion \
  && update-ca-certificates
 
-ARG JP_VERSION="0.1.3"
+ARG JP_VERSION="0.2.1"
```
Comment (bebound) [inline Dockerfile:45]: There is no breaking change from 0.1.3 to 0.2.1
0.2.0 add `jp-linux-arm64` release.

File: Dockerfile [hunk 2]
```diff
@@ -42,9 +42,9 @@ RUN apk add --no-cache bash openssh ca-certificates jq curl openssl perl git zip
...
 
-RUN curl -L https://github.com/jmespath/jp/releases/download/${JP_VERSION}/jp-linux-amd64 -o /usr/local/bin/jp \
+RUN arch=$(arch | sed s/aarch64/arm64/ | sed s/x86_64/amd64/) && curl -L https://github.com/jmespath/jp/releases/download/${JP_VERSION}/jp-linux-$arch -o /usr/local/bin/jp \
```
Comment (jiasli) [inline Dockerfile:47]: It's actually bad practice to include additional tools in the docker image. The image should be as small as possible.

Comment (bebound) [reply Dockerfile:47]: I have no idea why this image includes so many unrelated packages.
The image for Azure Linux is much smaller: https://github.com/Azure/azure-cli/pull/27204

Comment (jiasli) [inline Dockerfile:47]: Using `sed` to do the conversion is clever. I also saw this approach in https://stackoverflow.com/a/70369688/2199657.

**Review by jiasli — APPROVED** [review]

**Pull request merged**